### ISSUE TITLE
Browserify transform in package.json causes build errors

### ISSUE THREAD

**Issue by idolize** [ubilabs/react-geosuggest#11]
I am using Browserify to build my own project, but my build fails when I pull in this project as a dependency:

```
Cannot find module 'reactify' from '/node_modules/react-geosuggest'
```
This is because you have [`reactify` in the browserify transforms section of your `package.json`](https://github.com/ubilabs/react-geosuggest/blob/master/package.json#L28) (which means it is always executed when Browserify runs, even when it is `required` in by outside projects that install it via npm), but it isn't installed via `dependencies`. Thus, it crashes when Browserify can't find the missing dependency.

From the [Browserify docs](https://github.com/substack/node-browserify#browserifytransform):

> Now (that you have added a transform field to package.json) when somebody require()s your module, (the transform) will automatically be applied to the files in your module without explicit intervention by the person using your module.

I don't think this is a good practice for modules which are published for use on npm, because the compilation should have happened before publishing anyway (as you are already doing).

Since you are already using a Gulpfile, perhaps it would make more sense to just perform the JSX transformation as part of your Gulp task?

*(FYI, this was working for me on version 1.2.1, but after the release of 1.2.3 it is now breaking)*

**Comment by ro-ka** [comment]
Thanks, I knew there was a reason! Reverted it for now, but will leave this ticket open and review the build process later, no time right now. New (fixed) release is `1.3.0` on npm.

**Comment by zallek** [comment]
Adding `reactify` in `dependencies` leads that the package will be included in the final bundle whereas it's only needed for compilation. 
Was working fine with `reactify` in `devDependencies` with **webpack** at least.

**Comment by ro-ka** [comment]
Already moved it back to the `dependencies`, update to `1.3.0`. Will verify the build process and check how to preprocess this later.

**Comment by idolize** [comment]
@ro-ka Thanks!

@zallek Yes, having it in `devDependencies` is best, but the problem is that it will fail hard for anyone bundling their code with Browserify (webpack will work fine, but clearly not everyone out there uses the same build tools). Keeping it in `dependencies` is just a temporary workaround for a larger problem.

The best solution is to take the transform out of `package.json`, so Browserify never attempts to perform a transform after downloading it from npm (that feature is not needed here). The transform can still be performed as part of the build before publishing to npm through the `prepublish` section in `scripts` of `package.json`, or as part of the Gulp build task—at this point it will be safe to keep `reactify` in `devDependencies` instead of `dependencies`.

**Comment by idolize** [comment]
@ro-ka If you are looking for a good JSX transpilation tool to add to `prepublish`, I'd recommend [Babel](https://babeljs.io/docs/usage/cli/).

This script should replicate your current build structure without using `reactify`:

```js
"scripts": {
  "prepublish": "babel ./src --out-dir ./dist",
},
"devDependencies": {
  "babel": "*"
}
```

Also, as an aside, you may want to add a [`.npmignore`](https://docs.npmjs.com/misc/developers#keeping-files-out-of-your-package) file to keep the `src` directory out of npm. [More info on this subject here](http://stackoverflow.com/questions/13645824/can-i-write-npm-package-in-coffeescript).

**Comment by ro-ka** [comment]
I know, using babel in many other projects. Just no time right now to change the build process. Will do so next week, hopefully!

**Comment by ro-ka** [comment]
Fixed and released as `1.4.0`.

**Comment by idolizesc** [comment]
@ro-ka Thanks! Unfortunately, I'm getting errors trying to `require` this now with `1.4.0` released. First glance looks like it's because the `main` section of `package.json` points to `"module/Geosuggest.js"`, but I don't see this file in my latest npm install.

Are you sure all compiled files were included in your `npm publish`?

**Comment by idolize** [comment]
@ro-ka Thanks! Unfortunately, I'm getting errors trying to `require` this now with `1.4.0` released. First glance looks like it's because the `main` section of `package.json` points to `"module/Geosuggest.js"`, but I don't see this file in my latest npm install.

Are you sure all compiled files were included in your `npm publish`?

**Comment by ro-ka** [comment]
Ah… https://docs.npmjs.com/misc/developers#keeping-files-out-of-your-package
Will fix, sorry!

**Comment by ro-ka** [comment]
Fixed in `1.4.1`. Thanks for the feedback! Does it work now?

**Comment by idolize** [comment]
@ro-ka I updated and now I'm seeing a different error:
```
Cannot find module './GeosuggestItem.jsx' from '/node_modules/react-geosuggest/module'
```
